tuf/signed/verify.go

```diff
@@ -107,3 +108,30 @@ func VerifySignature(msg []byte, sig *data.Signature, pk data.PublicKey) error {
 	sig.IsValid = true
 	return nil
 }
+
+// VerifyPublicKeyMatchesPrivateKey checks if the private key and the public keys forms valid key pairs.
+// This should work with both ecdsa-x509 certificate PublicKey as well as ecdsa PublicKey
+func VerifyPublicKeyMatchesPrivateKey(privKey data.PrivateKey, pubKey data.PublicKey) error {
```

Comment:
Yes I agree! I have come across `utils.CanonicalKeyID()` and thought maybe that will be a good way of comparing key pair since it will extract the key id of the public key embedded in the certificate which should match privateKey.ID().